packages/python-packages/apiview-stub-generator/tests/apiview_test.py

```diff
@@ -14,57 +14,55 @@
 class TestApiView:
     def _count_newlines(self, apiview: ApiView):
         newline_count = 0
-        for token in apiview.tokens[::-1]:
-            if token.kind == TokenKind.Newline:
+        for line in apiview.review_lines[::-1]:
+            if len(line.tokens) == 0:
                 newline_count += 1
             else:
                 break
         return newline_count
 
     # Validates that there are no repeat defintion IDs and that each line has only one definition ID.
-    def _validate_definition_ids(self, apiview: ApiView):
-        definition_ids = set()
-        def_ids_per_line = [[]]
-        index = 0
-        for token in apiview.tokens:
-            # ensure that there are no repeated definition IDs.
-            if token.definition_id:
-                if token.definition_id in definition_ids:
-                    fail(f"Duplicate defintion ID {token.definition_id}.")
-                definition_ids.add(token.definition_id)
-            # Collect the definition IDs that exist on each line
-            if token.definition_id:
-                def_ids_per_line[index].append(token.definition_id)
-            if token.kind == TokenKind.Newline:
-                index += 1
-                def_ids_per_line.append([])
-        # ensure that each line has either 0 or 1 definition ID.
-        failures = [row for row in def_ids_per_line if len(row) > 1]
-        if failures:
-            fail(f"Some lines have more than one definition ID. {failures}")
-        
+    def _validate_line_ids(self, apiview: ApiView):
+        line_ids = set()
+
+        def collect_line_ids(review_lines, index=0):
+            for line in review_lines:
+                # Ensure that each line has either 0 or 1 definition ID.
+                if line.line_id and not isinstance(line.line_id, str):
+                    fail(f"Some lines have more than one definition ID. {line.line_id}")
+                # Ensure that there are no repeated definition IDs.
+                if line.line_id and line.line_id in line_ids:
+                    fail(f"Duplicate definition ID {line.line_id}.")
+                    line_ids.add(line.line_id)
+                # Recursively collect definition IDs from child lines
+                if line.children:
+                    collect_line_ids(line.children, index)
+
```

Comment:
I would recommend adding logic that validates the behavior of `related_to_line` and `is_context_end` rather than relying on manual testing. 

Here's how I did it in Swift:
https://github.com/Azure/azure-sdk-tools/pull/9506/files#r1904676885

And TypeSpec:
https://github.com/Azure/azure-sdk-tools/pull/9367/files#r1904677876

Reply:
updated!